core/src/ast/utils.rs

```diff
@@ -21,3 +24,224 @@ pub fn get_doc_lines(attrs: &[Attribute]) -> String {
 
     lines
 }
+
+#[derive(Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+pub struct RustLink {
+    path: Path,
+    typ: DocType,
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+enum DocType {
+    Struct,
+    StructField,
+    Enum,
+    EnumVariant,
+    EnumVariantField,
+    Trait,
+    FnInStruct,
+    FnInEnum,
+    FnInTrait,
+    DefaultFnInTrait,
+    Fn,
+    Mod,
+    Constant,
+    Macro,
+}
+
+pub fn get_rust_link(attrs: &[Attribute]) -> Option<RustLink> {
+    for attr in attrs {
+        if attr.path.to_token_stream().to_string() == "diplomat :: rust_link" {
+            if let Ok(syn::Meta::List(syn::MetaList { nested, .. })) = attr.parse_meta() {
+                if nested.len() == 2 {
+                    if let (
+                        syn::NestedMeta::Meta(syn::Meta::Path(path)),
+                        syn::NestedMeta::Meta(syn::Meta::Path(typ)),
+                    ) = (&nested[0], &nested[1])
+                    {
+                        if let Some(typ) = typ.get_ident() {
+                            return Some(RustLink {
+                                path: Path::from_syn(path),
+                                typ: match typ.to_token_stream().to_string().as_str() {
+                                    "Struct" => DocType::Struct,
+                                    "StructField" => DocType::StructField,
+                                    "Enum" => DocType::Enum,
+                                    "EnumVariant" => DocType::EnumVariant,
+                                    "EnumVariantField" => DocType::EnumVariantField,
+                                    "Trait" => DocType::Trait,
+                                    "FnInStruct" => DocType::FnInStruct,
+                                    "FnInEnum" => DocType::FnInEnum,
+                                    "FnInTrait" => DocType::FnInTrait,
+                                    "DefaultFnInTrait" => DocType::DefaultFnInTrait,
+                                    "Fn" => DocType::Fn,
+                                    "Mod" => DocType::Mod,
+                                    "Constant" => DocType::Constant,
+                                    "Macro" => DocType::Macro,
+                                    x => panic!("Invalid doc type {:?}", x),
+                                },
+                            });
+                        }
+                    }
+                }
+            }
+            panic!("Malformed attribute: {}", attr.to_token_stream());
+        }
+    }
+
+    None
+}
+
+impl RustLink {
+    pub fn http(&self, base_urls: &HashMap<String, String>) -> String {
```

Comment:
thought: In most cases we'll want to link to unicode-org.github.io, I wonder if we should instead support a _single_ base URL, with std/etc defaulting to doc.rust-lang.org and everything else going there. We can expand on that later.

People can set that URL to docs.rs/latest if they'd like

Reply:
We can't default a single base URL to `http://docs.rs/latest`, because the syntax is `http://docs.rs/<crate>/latest`

Reply:
Yeah we'll need special-casing with docs.rs. I'm okay with that.